src/EventListener/HeaderReplay/BackendSessionListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener\HeaderReplay;
+
+use Symfony\Component\HttpFoundation\Request;
+use Terminal42\HeaderReplay\Event\HeaderReplayEvent;
+use Terminal42\HeaderReplay\EventListener\HeaderReplayListener;
+
+/**
+ * Disables the reverse proxy based on the terminal42/header-replay-bundle.
+ *
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+class BackendSessionListener
+{
+    /**
+     * @var bool
+     */
+    private $disableIpCheck;
+
+    /**
+     * BackendSessionListener constructor.
+     *
+     * @param bool $disableIpCheck
+     */
+    public function __construct($disableIpCheck)
+    {
+        $this->disableIpCheck = $disableIpCheck;
+    }
+
+    /**
+     * @param HeaderReplayEvent $event
+     */
+    public function onReplay(HeaderReplayEvent $event)
+    {
+        $request = $event->getRequest();
```

Comment:
Same applies here. The event listener should only be triggered on `$request->attributes->get('_scope') === 'backend'` (actually use the constant!)